rust/cbork-utils/src/deterministic_helper.rs

```diff
@@ -0,0 +1,933 @@
+/// - The input uses indefinite length encoding (additional info = 31)
+/// - The additional information value is invalid
+pub fn get_cbor_header_size(bytes: &[u8]) -> Result<usize, minicbor::decode::Error> {
+    // Check if input is empty, which is invalid CBOR
+    if bytes.is_empty() {
+        minicbor::decode::Error::message("Empty cbor bytes");
+    }
+
+    // Extract the first byte which contains both major type and additional info
+    let first_byte = bytes
+        .first()
+        .copied()
+        .ok_or_else(|| minicbor::decode::Error::message("Empty cbor data"))?;
+    // Major type is in the high 3 bits (not used in this function but noted for clarity)
+    // let major_type = first_byte >> 5;
+    // Additional info is in the low 5 bits and determines header size
+    let additional_info = first_byte & 0b0001_1111;
+
+    // Calculate header size based on additional info value
+    match additional_info {
+        // Values 0-23 are encoded directly in the additional info bits
+        // Header is just the initial byte
+        0..=23 => Ok(1),
+
+        // Value 24 means the actual value is in the next 1 byte
+        // Header is 2 bytes (initial byte + 1 byte)
+        24 => Ok(2),
+
+        // Value 25 means the actual value is in the next 2 bytes
+        // Header is 3 bytes (initial byte + 2 bytes)
+        25 => Ok(3),
+
+        // Value 26 means the actual value is in the next 4 bytes
+        // Header is 5 bytes (initial byte + 4 bytes)
+        26 => Ok(5),
+
+        // Value 27 means the actual value is in the next 8 bytes
+        // Header is 9 bytes (initial byte + 8 bytes)
+        27 => Ok(9),
+        // Value 31 indicates indefinite length, which is not allowed in
+        // deterministic encoding per RFC 8949 section 4.2.1
+        31 => {
+            Err(minicbor::decode::Error::message(
+                "Cannot determine size of indefinite length item",
+            ))
+        },
+
+        // Values 28-30 are reserved in RFC 8949 and not valid in current CBOR
+        _ => {
+            Err(minicbor::decode::Error::message(
+                "Invalid additional info in CBOR header",
+            ))
+        },
+    }
+}
+
+/// Extracts a slice of raw CBOR bytes from the decoder's input buffer for a given range.
+///
+/// This function safely extracts bytes between the specified start and end positions,
+/// performing bounds checking to ensure the requested range is valid.
+fn extract_cbor_bytes(
+    decoder: &minicbor::Decoder<'_>, range_start: usize, range_end: usize,
+) -> Result<Vec<u8>, minicbor::decode::Error> {
+    // Validate CBOR byte range bounds
+    if range_start >= range_end {
+        minicbor::decode::Error::message(
+            "Invalid CBOR byte range: start must be less than end position",
+        );
+    }
+
+    let byte_length = range_end.saturating_sub(range_start);
+    get_checked_slice(decoder.input(), range_start, byte_length).map(<[u8]>::to_vec)
```

Comment:
this could be replaced with the `get_map_bytes`

Reply:
resolved - good catch